homeassistant/setup.py

```diff
@@ -219,7 +220,7 @@ async def _async_process_dependencies(
     return failed
 
 
-async def _async_setup_component(
+async def _async_setup_component(  # noqa: C901
```

Comment:
I hate to add a C901 here but I didn't want to break this up in this PR as it was make it much harder to review